frontend/src/components/ui/PipelineDeployButton.jsx

```diff
@@ -1,40 +1,13 @@
 import { useMutation } from "@tanstack/react-query";
-import { Deploy, SubtractAlt } from "@carbon/icons-react";
+import { Deploy } from "@carbon/icons-react";
 import { Button } from "@carbon/react";
 import { deployPipeline } from "@/client/anvil";
-import { useQueryClient } from "@tanstack/react-query";
-
-export const PipelineDeployButton = ({
-  deployed,
-  uuid,
-  hash,
-  configuration,
-}) => {
-  // const queryClient = useQueryClient();
 
+export const PipelineDeployButton = ({ uuid, hash, configuration }) => {
   const deploy = useMutation({
     mutationFn: async () => {
       return await deployPipeline(configuration, uuid, hash);
     },
-    onSuccess: (data) => {
-      // Update the React Query cache
-      console.log(data);
-      const queryKey = ["pipelines", configuration?.anvil?.host];
-      queryClient.setQueryData(queryKey, (pipelines) => {
-        return pipelines?.body.map((pipeline) => {
-          if (pipeline.Hash === hash) {
-            console.log("updating ", hash);
-            // Update the deployed status of the matching pipeline
-            const newPipeline = {
-              ...pipeline,
-              Deployed: true,
-            };
-            return newPipeline;
-          }
-          return pipeline;
-        });
-      });
-    },
   });
```

Comment:
Extra attention here as well. `OnSuccess` would throw since `queryClient` is undefined. I would like confirmation that this code is not needed anymore.

Reply:
Oh, this code is definitely still real, I'm not sure why queryClient is commented out

Reply:
I tried to fix this but because it's on your remote it's difficult for me to get the changes pushed